### PR TITLE
Add new tab completion feature

### PR DESCRIPTION
**Pull request by b4b4r07**
## WHAT

Introduce tab completion by `$ENHANCD_FILTER` and it's inspired by [changyuheng/zsh-interactive-cd](https://github.com/changyuheng/zsh-interactive-cd)

## WHY

To fix https://github.com/b4b4r07/enhancd/issues/87

### CODE REVIEW

File: src/completion.zsh [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+__enhancd::completion::run() {
+  setopt localoptions noshwordsplit noksh_arrays noposixbuiltins
+  local tokens cmd
+
+  tokens=(${(z)LBUFFER})
+  cmd=${tokens[1]}
+
+  if [[ "$LBUFFER" =~ "^\ *cd$" ]]; then
```
Comment (Konfekt) [inline src/completion.zsh]: Awesome work, but this regular expression is too restrictive. In particular, it should not check for `cd` but `$ ENHANCD_COMMAND`.

File: src/completion.zsh [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    zle ${ENHANCD_COMPLETION_DEFAULT:-expand-or-complete}
+  elif [ "$cmd" = cd ]; then
```
Comment (Konfekt) [inline src/completion.zsh]: Same here, it should check for `$ENHANCD_COMMAND`.

File: src/completion.zsh (hunk 1 above)
Comment (b4b4r07) [reply src/completion.zsh]: Nice catch!! I'll update those 👍 @Konfekt

Comment (b4b4r07) [reply src/completion.zsh]: Done 7f6dcf0973b5d448ecb8adebda96f28024965249

**Pull request merged**